src/service/CompasSclDataService.ts

```diff
@@ -0,0 +1,256 @@
+import { formatXml } from '../file.js';
+
+import {
+  extractSclFromResponse,
+  getWebsocketUri,
+  handleError,
+  handleResponse,
+  parseXml,
+} from './foundation.js';
+import { websocket } from './Websocket.js';
+
+export const SDS_NAMESPACE =
+  'https://www.lfenergy.org/compas/SclDataService/v1';
+
+export enum ChangeSet {
+  MAJOR = 'MAJOR',
+  MINOR = 'MINOR',
+  PATCH = 'PATCH',
+}
+
+export interface CreateRequestBody {
+  sclName: string;
+  comment: string | null;
+  doc: Document;
+}
+
+export interface UpdateRequestBody {
+  changeSet: ChangeSet;
+  comment: string | null;
+  doc: Document;
+}
+
+export function CompasSclDataService() {
```

Comment:
instead of the function CompasSclDataService,
you could use ts namespaces
https://www.typescriptlang.org/docs/handbook/namespaces.html

```ts

export namespace CompasSclDataService {

  export function listSclTypes(): Promise<Document> {
    ...
  }
}
```

```ts
import { SclDataService } from './scl-data-service.js';

SclDataService.listSclTypes().then((res) => ... );
```

Not better or worse, just different taste :)

Reply:
Not changing this currrently